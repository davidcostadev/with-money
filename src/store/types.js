export const ADD_USER = 'ADD_USER';
export const TO_CURRENT_MONTH = 'TO_CURRENT_MONTH';
export const TO_PREVIUS_MONTH = 'TO_PREVIUS_MONTH';
export const TO_NEXT_MONTH = 'TO_NEXT_MONTH';
