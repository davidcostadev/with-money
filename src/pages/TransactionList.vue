<template>
  <div>
    <toolbar v-on:update:stateMonth="getTransactions" />
    <md-tabs>
      <md-tab md-label="Fixas"></md-tab>
      <md-tab md-label="VariÃ¡veis"></md-tab>
    </md-tabs>
    <md-table>
      <md-table-row>
        <md-table-head>Nome</md-table-head>
        <md-table-head>Valor</md-table-head>
        <md-table-head>Pago?</md-table-head>
        <md-table-head>Data</md-table-head>
      </md-table-row>
      <md-table-row
        v-for="transaction in transactions"
        :key="transaction.id"
        @click="onSelectTransaction(transaction)"
      >
        <md-table-cell>{{transaction.name}}</md-table-cell>
        <md-table-cell>{{transaction.value}}</md-table-cell>
        <md-table-cell>{{transaction.transationDate}}</md-table-cell>
        <md-table-cell>{{transaction.isPaid ? 'Paid' : 'Not Paid'}}</md-table-cell>
      </md-table-row>
    </md-table>
  </div>
</template>

<script>
import Moment from 'moment';
import { mapGetters } from 'vuex';
import Transaction from '../services/transactions';
import Toolbar from '../components/Toolbar';

export default {
  components: { Toolbar },
  data() {
    return {
      transactions: [],
    };
  },
  computed: {
    ...mapGetters(['state_month']),
  },
  methods: {
    onSelectTransaction({ id }) {
      this.$router.push(`/transaction/${id}`);
    },
    async getTransactions() {
      const start = Moment(this.state_month).startOf('month').toISOString();
      const end = Moment(this.state_month).endOf('month').toISOString();
      const { data } = await Transaction.getTransactions({
        createdAt: [
          start,
          end,
        ].join(','),
      });

      this.transactions = data;
    },
  },
  created() {
    this.getTransactions();
  },
};
</script>

<style>
  .md-content {
    padding: 16px;
  }
</style>
