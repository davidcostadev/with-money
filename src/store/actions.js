import * as types from './types';

export default {
  addUser({ commit }, payload) {
    commit(types.ADD_USER, payload);
  },
  toNextMonth({ commit }) {
    commit(types.TO_NEXT_MONTH);
  },
  toCurrentMonth({ commit }) {
    commit(types.TO_CURRENT_MONTH);
  },
  toPreviousMonth({ commit }) {
    commit(types.TO_PREVIUS_MONTH);
  },
};
