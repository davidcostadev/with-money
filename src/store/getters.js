export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
