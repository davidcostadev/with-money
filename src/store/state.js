import Moment from 'moment';

export default {
  user: null,
  current_month: Moment(),
  state_month: Moment(),
  state_month_str: Moment().format('MMM YY'),
  next_month_str: Moment().add(1, 'month').format('MMM YY'),
  previous_month_str: Moment().subtract(1, 'month').format('MMM YY'),
  flash_message: '',
};
