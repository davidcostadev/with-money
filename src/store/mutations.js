import Moment from 'moment';
import * as types from './types';
import { updateStatesMont } from '../utils/state';

export default {
  [types.ADD_USER](state, payload) {
    state.user = payload;
  },
  [types.TO_CURRENT_MONTH](state) {
    state.state_month = state.current_month;
    updateStatesMont(state);
  },
  [types.TO_PREVIUS_MONTH](state) {
    state.state_month = Moment(state.state_month).subtract(1, 'month');
    updateStatesMont(state);
  },
  [types.TO_NEXT_MONTH](state) {
    state.state_month = Moment(state.state_month).add(1, 'month');
    updateStatesMont(state);
  },
};
