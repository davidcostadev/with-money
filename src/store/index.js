import Vue from 'vue';
import Vuex from 'vuex';
import Moment from 'moment';
import mutations from './mutations';
import actions from './actions';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    user: null,
    current_month: Moment(),
    state_month: Moment(),
    state_month_str: Moment().format('MMMM YY'),
    next_month_str: Moment().add(1, 'month').format('MMMM YY'),
    previous_month_str: Moment().subtract(1, 'month').format('MMMM YY'),
  },
  getters: {
    user(state) {
      return state.user;
    },
    current_month(state) {
      return state.current_month;
    },
    state_month(state) {
      return state.state_month;
    },
    state_month_str(state) {
      return state.state_month_str;
    },
    next_month_str(state) {
      return state.next_month_str;
    },
    previous_month_str(state) {
      return state.previous_month_str;
    },
  },
  mutations,
  actions,
});
