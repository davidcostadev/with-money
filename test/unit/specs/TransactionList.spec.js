import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import VueMaterial from 'vue-material';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import TransactionList from '../../../src/pages/TransactionList';
import store from '../../../src/store';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueMaterial);
localVue.use(VueRouter);

const transactions = [
  {
    id: 3,
    AccountId: 1,
    name: 'AlmoçoAlmoço 2',
    value: '15.00',
    type: 'out',
    isPaid: false,
    transactionDate: '2018-07-02',
    createdAt: '2018-08-04T21:26:56.000Z',
    updatedAt: '2018-08-04T21:27:07.000Z',
  },
  {
    id: 2,
    AccountId: 2,
    name: 'Coxinha',
    value: '3.13',
    type: 'out',
    isPaid: false,
    transactionDate: '2018-08-04',
    createdAt: '2018-08-04T19:08:54.000Z',
    updatedAt: '2018-08-04T19:08:54.000Z',
  },
  {
    id: 1,
    AccountId: 1,
    name: 'Lanche na Sonia',
    value: '30.90',
    type: 'out',
    isPaid: true,
    transactionDate: '2018-08-04',
    createdAt: '2018-08-04T19:08:54.000Z',
    updatedAt: '2018-08-04T19:08:54.000Z',
  },
];

const mock = {
  data: transactions,
  pagination: {
    totalItems: 3,
    currentPage: 1,
    perPage: 100,
    totalPages: 1,
    nextPage: null,
    previousPage: null,
  },
};

jest.mock('../../../src/services/transactions', () => ({
  getTransactions: () => Promise.resolve(mock),
}));

describe('TransactionList Component', () => {
  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(TransactionList, { store, localVue });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    it('with transactions', () => {
      const wrapper = mount(TransactionList, { store, localVue });
      wrapper.setData({
        transactions,
      });
      // wrapper.vm.$forceUpdate();

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
