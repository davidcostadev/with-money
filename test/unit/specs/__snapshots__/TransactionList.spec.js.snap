// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList Component Render default 1`] = `
<div
  class="full-height"
>
  <toolbar-stub />
   
  <mdtabs-stub
    id="tabs-type"
  >
    <mdtab-stub />
     
    <mdtab-stub />
  </mdtabs-stub>
   
  <mdtable-stub>
    false
  </mdtable-stub>
   
  <mdbutton-stub
    class="md-fab md-primary menu-speed"
  >
    <mdicon-stub>
      note_add
    </mdicon-stub>
  </mdbutton-stub>
</div>
`;

exports[`TransactionList Component Render with transactions 1`] = `
<div
  class="full-height"
>
  <div
    class="md-toolbar md-theme-default md-elevation-4"
  >
    <button
      class="md-button md-theme-default"
      id="btn-previus-month"
      type="button"
    >
      <div
        class="md-ripple"
      >
        <div
          class="md-button-content"
        >
          <i
            class="md-icon md-icon-font md-theme-default"
          >
            chevron_left
          </i>
           July 18
  
        </div>
         
      </div>
    </button>
     
    <h3
      class="md-title"
      style="text-align: center;"
    >
      August 18
    </h3>
     
    <button
      class="md-button md-theme-default"
      id="btn-next-month"
      type="button"
    >
      <div
        class="md-ripple"
      >
        <div
          class="md-button-content"
        >
          <i
            class="md-icon md-icon-font md-theme-default"
          >
            chevron_right
          </i>
           September 18
  
        </div>
         
      </div>
    </button>
  </div>
   
  <div
    class="md-tabs md-alignment-left md-no-transition md-theme-default"
    id="tabs-type"
  >
    <div
      class="md-tabs-navigation md-elevation-0"
    >
       
      <span
        class="md-tabs-indicator"
      />
    </div>
     
    <div
      class="md-content md-tabs-content md-theme-default"
      style="height: 0px; display: none;"
    >
      <div
        class="md-tabs-container"
      >
        <div
          class="md-tab-stubbed"
          id="tab-in"
          md-label="Entradas"
        />
         
        <div
          class="md-tab-stubbed"
          id="tab-out"
          md-label="Saidas"
        />
      </div>
    </div>
  </div>
   
  <div
    class="md-card md-table md-theme-default"
    md-card=""
  >
     
    <!---->
     
    <!---->
     
    <div
      class="md-content md-table-content md-scrollbar md-theme-default"
    >
      <table>
        <!---->
         
        <tbody>
          <tr
            class="md-table-row"
          >
            <!---->
             
            <th
              class="md-table-head"
            >
              <div
                class="md-table-head-container"
              >
                <div
                  class="md-table-head-label"
                >
                  Data
                </div>
              </div>
            </th>
             
            <th
              class="md-table-head"
            >
              <div
                class="md-table-head-container"
              >
                <div
                  class="md-table-head-label"
                >
                  Nome
                </div>
              </div>
            </th>
             
            <th
              class="md-table-head"
            >
              <div
                class="md-table-head-container"
              >
                <div
                  class="md-table-head-label"
                >
                  Valor
                </div>
              </div>
            </th>
             
            <th
              class="md-table-head"
            >
              <div
                class="md-table-head-container"
              >
                <div
                  class="md-table-head-label"
                >
                  Categoria
                </div>
              </div>
            </th>
             
            <th
              class="md-table-head"
              style="width: 75px;"
            >
              <div
                class="md-table-head-container"
              >
                <div
                  class="md-table-head-label"
                >
                  Pago?
                </div>
              </div>
            </th>
          </tr>
           
          <tr
            class="md-table-row"
            id="transaction-3"
          >
            <!---->
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                07/02
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                AlmoçoAlmoço 2
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                15.00
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                Lanche
              </div>
            </td>
             
            <td
              class="md-table-cell md-table-cell-center"
            >
              <div
                class="md-table-cell-container"
              >
                <i
                  class="md-icon md-icon-font md-theme-default"
                >
                  lens
                </i>
              </div>
            </td>
          </tr>
          <tr
            class="md-table-row"
            id="transaction-2"
          >
            <!---->
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                08/04
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                Coxinha
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                3.13
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                Lanche
              </div>
            </td>
             
            <td
              class="md-table-cell md-table-cell-center"
            >
              <div
                class="md-table-cell-container"
              >
                <i
                  class="md-icon md-icon-font md-theme-default"
                >
                  lens
                </i>
              </div>
            </td>
          </tr>
          <tr
            class="md-table-row"
            id="transaction-1"
          >
            <!---->
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                08/04
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                Lanche na Sonia
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                30.90
              </div>
            </td>
             
            <td
              class="md-table-cell"
            >
              <div
                class="md-table-cell-container"
              >
                Lanche
              </div>
            </td>
             
            <td
              class="md-table-cell md-table-cell-center"
            >
              <div
                class="md-table-cell-container"
              >
                <i
                  class="md-icon md-icon-font md-theme-default"
                >
                  check_circle
                </i>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
       
    </div>
     
    <!---->
  </div>
   
  <a
    class="md-button md-fab md-primary menu-speed md-theme-default"
    href="#/transaction-new?type=out"
  >
    <div
      class="md-ripple"
    >
      <div
        class="md-button-content"
      >
        <i
          class="md-icon md-icon-font md-theme-default"
        >
          note_add
        </i>
      </div>
       
    </div>
  </a>
</div>
`;
